diffsims/tests/test_utils/test_sim_utils.py

```diff
@@ -132,3 +145,176 @@ def test_kinematic_simulator_invalid_illumination():
+        data = np.zeros((100, 100))
+        am = tesla_to_am(data)
+        assert data.shape == am.shape
+        assert (data == 0.00).all()
+
+    def test_ones(self):
+        data = np.ones((100, 100)) * 10
+        am = tesla_to_am(data)
+        assert data.shape == am.shape
+        assert (data != 0.0).all()
+
+    def test_known_value(self):
+        tesla = 1
+        am = tesla_to_am(tesla)
+        assert approx(tesla, rel=1e-4) == 795775
+
+
+class TestAccelerationVoltageToVelocity:
+    def test_zero(self):
+        assert acceleration_voltage_to_velocity(0) == 0.0
+
+    @pytest.mark.parametrize(
+        "av,vel", [(100000, 1.6434e8), (200000, 2.0844e8), (300000, 2.3279e8)]
+    )  # V, m/s
+    def test_values(self, av, vel):
+        v = acceleration_voltage_to_velocity(av)
+        assert approx(v, rel=0.001) == vel
+
+
+class TestAccelerationVoltageToRelativisticMass:
+    def test_zero(self):
+        mr = acceleration_voltage_to_relativistic_mass(0.0)
+        assert approx(mr) == sc.electron_mass
+
+    def test_200kv(self):
+        mr = acceleration_voltage_to_relativistic_mass(200000)
+        assert approx(mr) == 1.268e-30
+
+
+@pytest.mark.parametrize(
+    "av,wl", [(100000, 3.701e-12), (200000, 2.507e-12), (300000, 1.968e-12)]
+)  # V, pm
+def test_acceleration_voltage_to_wavelength(av, wl):
+    wavelength = acceleration_voltage_to_wavelength(av)
+    assert approx(wavelength, rel=0.001, abs=0.0) == wl
+
+
+def test_acceleration_voltage_to_wavelength_array():
+    av = np.array([100000, 200000, 300000])  # In Volt
+    wavelength = acceleration_voltage_to_wavelength(av)
+    wl = np.array([3.701e-12, 2.507e-12, 1.968e-12])  # In pm
+    assert len(wl) == 3
+    assert approx(wavelength, rel=0.001, abs=0.0) == wl
+
+
+class TestDiffractionScatteringAngle:
+    def test_simple(self):
+        # This should give ~9.84e-3 radians
+        acceleration_voltage = 300000
+        lattice_size = 2e-10  # 2 Ångstrøm (in meters)
+        miller_index = (1, 0, 0)
+        scattering_angle = diffraction_scattering_angle(
+            acceleration_voltage, lattice_size, miller_index
+        )
+        assert approx(scattering_angle, rel=0.001) == 9.84e-3
+
+    @pytest.mark.parametrize(
+        "mi,sa",
+        [
+            ((1, 0, 0), 9.84e-3),
+            ((0, 1, 0), 9.84e-3),
+            ((0, 0, 1), 9.84e-3),
+            ((2, 0, 0), 19.68e-3),
+            ((0, 2, 0), 19.68e-3),
+            ((0, 0, 2), 19.68e-3),
+        ],
+    )
+    def test_miller_index(self, mi, sa):
+        acceleration_voltage = 300000
+        lattice_size = 2e-10  # 2 Ångstrøm (in meters)
+        scattering_angle = diffraction_scattering_angle(
+            acceleration_voltage, lattice_size, mi
+        )
+        assert approx(scattering_angle, rel=0.001) == sa
+
+    def test_array_like(self):
+        # This should give ~9.84e-3 radians
+        acceleration_voltage = 300000
+        lattice_size = np.array([2e-10, 2e-10])
```

Comment:
could make these different? Only requires that you change the answers array a little further down.